sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -66,6 +65,12 @@ export class InteractiveBrowserCredential implements TokenCredential {
       this.redirectUri = "http://localhost";
     }
 
+    const url = new URL(this.redirectUri);
+    this.port = parseInt(url.port);
+    if (isNaN(this.port)) {
+      this.port = 80;
+    }
```

Comment:
In the odd scenario where somebody has the processor when this method is called, could it be possible for something actively reading this property to read it in a NaN state?

If that isn't plausible, then ignore this! 🌞

Reply:
Not sure what you're asking here. Though maybe this will help clarify the above, from `parseInt` docs: 

```
Return value
An integer parsed from the given string.

Or NaN when

the radix is smaller than 2 or bigger than 36, or
the first non-whitespace character cannot be converted to a number.
```

There may be clearer ways to do the above. I'm just detecting the NaN case and defaulting the port.